src/db.rs

```diff
@@ -64,8 +65,17 @@ impl Storage for DB {
         }
     }
 
-    fn remove(&mut self, _key: &[u8]) -> StdResult<()> {
-        // TODO: implement
-        dyn_contract_err("Not implemented".to_string())
+    fn remove(&mut self, key: &[u8]) -> StdResult<()> {
+        let key = Buffer::from_vec(key.to_vec());
+        // caller will free input
+        let go_result: GoResult = (self.vtable.remove_db)(self.state, key).into();
+        let _key = unsafe { key.consume() };
```

Comment:
>caller will free input

This is incorrect, the input is released after the call, on line 72

Reply:
Thanks

Reply:
I was a little blunt
The rest of the commit is :100:

Reply:
annddd i just realized it says "caller", not "callee". So it's actually correct...

Reply:
Copied it from set. Now removed it from both.Thanks